pkg/apis/core/types.go

```diff
@@ -3277,9 +3277,9 @@ type PodSpec struct {
 	// +optional
 	AutomountServiceAccountToken *bool
 
-	// NodeName is a request to schedule this pod onto a specific node.  If it is non-empty,
-	// the scheduler simply schedules this pod onto that node, assuming that it fits resource
-	// requirements.
+	// NodeName indicates in which node this pod is scheduled.
+	// If empty, this pod is a candidate for scheduling by the scheduler defined in schedulerName.
+	// If it is non-empty, the kubelet for this node becomes responsible for the lifecycle of this pod.
```

Comment:
Kubelet is always responsible for the lifecycle of the pod. If it is non empty, I thought it means that we skip the scheduling cycle for the pod.

Reply:
No to both.

The new comment is the accurate description.

Reply:
until nodeName is set, the kubelet isn't even aware of the pod's existence

once nodeName is set, kubelet takes over responsibility for running or rejecting the pod

Reply:
Let me give an example: it is possible to delete a Pod that is not scheduled, that is, a pod that has empty node name.

In that case, kubelet would have never looked at this Pod, even though it completed its lifecycle. In this case it's kcm which sets the Failed phase, if the Pod had a finalizer.

Reply:
Maybe I am taking "pod lifecycle" too literal. I was thinking https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/ where this is primarly the responsibility of Kubelet. Kubelet always works with a set nodeName regardless of whether a user sets NodeName or not.

Reply:
I think I get what you are saying..

Reply:
SGTM.